### ISSUE TITLE
Complete zipkin compatibility

### ISSUE THREAD

**Issue by adriancole** [Nike-Inc/wingtips#9]
Wingtips is already towards zipkin compatibility via use of the B3 headers. These ensure that trace data propagated in-band with requests would carry forward even if the next hop is golang or ruby.

The next steps is to ensure the out-of-band data can be sent to zipkin, so that those using wing-tips and the zipkin ui can actually see the timing data not sent in the B3 headers.

For example, this could be reporting via POST to a zipkin http server.

Once this is done, it would be great to share the news by updating http://zipkin.io/pages/existing_instrumentations.html with wingtips. This could be done via a pull request updating this file: https://github.com/openzipkin/openzipkin.github.io/blob/master/_data/existing_instrumentations.yml

**Comment by nicmunroe** [comment]
I've been working on this - should hopefully have a PR ready in a day or two.

**Comment by adriancole** [comment]
:applause:

**Comment by nicmunroe** [comment]
@adriancole PR is up. Please take a look and let me know if you see anything incorrect, or anything you think I should change to improve integration: https://github.com/Nike-Inc/wingtips/pull/10

**Comment by adriancole** [comment]
ps once things are released, not only would it be nice to have wingtips in the doc list, but also a working example. I've a couple here, which make a basic client+server showing how things work.

https://github.com/openzipkin/zipkin-finagle-example
https://github.com/openzipkin/sleuth-webmvc-example

This sort of project is a nice way to showcase the feature in a way even better than the README. If you feel up to making one, lemme know and I'll carve out a repo, like.. wingtips-servlet-example

**Comment by nicmunroe** [comment]
@adriancole yep I plan on doing a PR to get wingtips in the docs, and I was already thinking of doing an example project so that would be good too. I can see the example project being useful in both repos though so I'm not sure where it should go.

I'll try to get the docs PR up tomorrow sometime. The example project may take a while (other priorities unfortunately).

**Comment by nicmunroe** [comment]
Wingtips Zipkin compatibility has been added in the wingtips-zipkin module, and the Zipkin instrumentation page has been updated to include Wingtips. Closing this out (we can look at doing sample projects another time).

**Comment by adriancole** [comment]
thanks for all the work here!

**Comment by jpiechowka** [comment]
@nicmunroe @adriancole Do you think that my example wingtips-zipkin application is a good example to show how wingtips works?  [Wingtips Example](https://github.com/jpiechowka/wingtips-cujojs-spark-example)

**Comment by adriancole** [comment]
looks cool. I'd delete your repo and recreate it with a gitignore for
intellij and npm. That will reduce the clutter. We can take it from there
on the new repo?

Ex, you'll want a combo of these two

https://github.com/openzipkin/zipkin-js-example/blob/master/.gitignore
https://github.com/openzipkin/zipkin-finagle-example/blob/master/.gitignore

**Comment by jpiechowka** [comment]
@adriancole Sure, feel free to do anything with this code. I updated the server, so that it adds correct values in HTTP headers in the response.

**Comment by adriancole** [comment]
OK I'll probably squash this stuff into an openzipkin repo and give you
access (and credit!). will proceed tomorrow unless you yell no

**Comment by jpiechowka** [comment]
@adriancole Great, you can proceed! I will try my best to maintain it :D

**Comment by jpiechowka** [comment]
@adriancole I will clean the repo, add gitignore and reupload everything later today when I am at home, so it is easier to work with.

**Comment by adriancole** [comment]
if you want, just post it here.. just gave you write.

https://github.com/openzipkin/wingtips-cujojs-spark-example

welcome to zipkin!

**Comment by jpiechowka** [comment]
@adriancole Great! I can see you have created gitignore,

**Comment by jpiechowka** [comment]
@adriancole @nicmunroe 
I have uploaded files for the Wingtips example. @nicmunroe As soon as there will be Maven package with the new changes I will update the code.

**Comment by nicmunroe** [comment]
@jpiechowka wingtips version 0.11.0 is available and should fix the problems you were experiencing. Let me know how it goes!

**Comment by jpiechowka** [comment]
@nicmunroe @adriancole  I have tested it and Wingtips is generating correct Zipkin ID's and it is correctly parsing ID's from strings. I will update my example as soon as possible with the newer Maven packages.
